meilisearch-release-changelog/v1.10.0.md

```diff
@@ -0,0 +1,357 @@
+
+```bash
+curl \
+ -X POST http://localhost:7700/indexes/movies/search \
+  -H 'Content-Type: application/json' \
+ --data-binary '{
+  "q": "進撃の巨人",
+  "locales": ["jpn"]
+}'
+```
+
+The `locales` parameter overrides eventual `locales` in the index settings.
+
+## Experimental: edit documents by using a function
+
+You can edit documents by executing a [Rhai function](https://rhai.rs/) on all the documents of your database or a subset of them that you can select by a Meilisearch filter.
+
+Activate the experimental feature:
+
+```bash
+curl \
+ -X PATCH 'http://localhost:7700/experimental-features/' \
+  -H 'Content-Type: application/json' \
+ --data-binary '{
+    "editDocumentsByFunction": true
+ }'
+```
+
+By indexing this [movies dataset](https://www.meilisearch.com/movies.json), you can run the following Rhai function on all of them. This function will uppercase the titles of the movies with an `id` > 3000 and add sparkles around it. Rhai templating syntax is applied here:
+
+```bash
+curl http://localhost:7700/indexes/movies/documents/edit \
+ -H 'content-type: application/json' \
+  -d '{
+ "filter": "id > 3000",
+ "function": "doc.title = `✨ ${doc.title.to_upper()} ✨`"
+ }'
+```
+
+📖 More information [here](https://meilisearch.notion.site/Update-Documents-by-Function-0cff8fea7655436592e7c8a6de932062).
+
+🗣️ This is an experimental feature and we need your help to improve it! Share your thoughts and feedback on this [GitHub discussion](https://github.com/orgs/meilisearch/discussions/762).
+
+Done by @Kerollmops in #4626.
+
+## Experimental AI-powered search: quality of life improvements
+
+For the purpose of future stabilization of the feature, we are applying changes and quality-of-life improvements.
+
+Done by @dureuill in #4801, #4815, #4818, #4822.
+
+### ⚠️ Breaking changes - changing the parameters of the REST API
+
+The old parameters of the REST API are too numerous and confusing.
+
+Removed parameters: `query` , `inputField`, `inputType`, `pathToEmbeddings` and `embeddingObject`.
+Replaced by
+- `request` : A JSON value that represents the request made by Meilisearch to the remote embedder. The text to embed must be replaced by the placeholder value `“{{text}}”`.
+- `response`: A JSON value that represents a fragment of the response made by the remote embedder to Meilisearch. The embedding must be replaced by the placeholder value `"{{embedding}}"`.
+
+Before:
+
+```json5
+// v1.9 (old) version
+{
+  "source": "rest",
+  "url": "https://localhost:10006",
+  "query": {
+    "model": "minillm",
+ },
+  "inputField": ["prompt"],
+  "inputType": "text",
+  "embeddingObject": ["embedding"]
+}
+```
+
+```json5
+// v1.10 (new) version
+{
+  "source": "rest",
+  "url": "https://localhost:10006",
+  "request": {
+    "model": "minillm",
+    "prompt": "{{text}}"
+ },
+  "response": {
+    "embedding": "{{embedding}}"
+ }
+}
+```
+
```

Comment:
I think that we should highlight that this breaking change will make it impossible to import dumps or upgrade a Meilisearch instance on the Cloud when it has a REST embedder configured.

The paragraph I had in my PR seems apt

```suggestion
> [!CAUTION]
> This is a breaking change to the configuration of REST embedders.
> Importing a dump containing a REST embedder configuration will fail in v1.10 with an error: "Error: unknown field `query`, expected one of `source`, `model`, `revision`, `apiKey`, `dimensions`, `documentTemplate`, `url`, `request`, `response`, `distribution` at line 1 column 752".

Upgrade procedure (Cloud):

1. Remove any embedder with source "rest"
2. Click the button to upgrade your Meilisearch version
4. Re-add any removed embedder, using the new settings.

Upgrade procedure (OSS users):

1. Remove any embedder with source "rest"
2. Create a dump
3. Import that dump in a v1.10
4. Re-add any removed embedder, using the new settings.

```

Rendered below:

--- 

> [!CAUTION]
> This is a breaking change to the configuration of REST embedders.
> Importing a dump containing a REST embedder configuration will fail in v1.10 with an error: "Error: unknown field `query`, expected one of `source`, `model`, `revision`, `apiKey`, `dimensions`, `documentTemplate`, `url`, `request`, `response`, `distribution` at line 1 column 752".

Upgrade procedure (Cloud):

1. Remove any embedder with source "rest"
2. Click the button to upgrade your Meilisearch version
4. Re-add any removed embedder, using the new settings.

Upgrade procedure (OSS users):

1. Remove any embedder with source "rest"
2. Create a dump
3. Import that dump in a v1.10
4. Re-add any removed embedder, using the new settings.

Reply:
We could make it simpler and less heavy by linking the docs for steps after step 1

I will update